magpi_scraper.py

```diff
@@ -59,6 +72,19 @@ def downloadMagPi(URL):
 get_page_urls()
 
 # Begin threading process
+
+# COMMENT:
+#
+# If I understand this correctly, are you creating a thread for each
+# page and then having each thread download all the pdfs on its
+# particular page?
+#
+# This works, but if there are a lot of pages then you will spawn a
+# lot of threads. You have to be careful you don't create a denial of
+# service (DOS) attack unintentionally to a website.
+#
+# Sometimes a simple loop on the main thread works just fine and maybe
+# even faster depending on your computer and the website.
```

Comment:
Some concerns about Denial of Service (DOS) and threads.

Reply:
Ok, this is great advice; I was wondering whether I should use a loop instead of threading; I went with this threading format because I was able to get it to work first; Will see if I can get this scraper to work with a loop as well; Thanks